@market/components/NFTCardMarket.tsx

```diff
@@ -1,21 +1,28 @@
-import { useMemo } from 'react'
-import { NFTObject } from '@zoralabs/nft-hooks/dist/types/NFTInterface'
-import { useAskHelper, useRelevantMarket } from '@market/hooks'
 import { FillV3AskModal } from '@market/components'
-import { FlexProps } from '@zoralabs/zord'
+import { useAskHelper, useRelevantMarket } from '@market/hooks'
 import { UniversalListAskModal } from '@market/modules/PrivateAsk/UniversalListAskModal'
+import { useNFTProvider } from '@shared'
+import { NFTObject } from '@zoralabs/nft-hooks'
+import { FlexProps } from '@zoralabs/zord'
+
+export interface NFTCardMarketProps extends FlexProps {}
 
-export interface NFTCardMarketProps extends FlexProps {
-  nftObj: NFTObject
+export function NFTCardMarket(props: NFTCardMarketProps) {
+  const { nft: nftObj } = useNFTProvider()
```

Comment:
This pattern feels really weird to me. We already have token data for each token being fetched in order to lay out the NFT grid, and should ideally be making use of that data and potentially passing it in as initialData to the hook that fetches via SWR (which I'm assuming is called within NFTProvider?)

We shouldn't let this hold us back if it requires a significant refactor but we definitely don't want to be making individual fetches for each NFT card.